.github/workflows/shared-build-deploy-container.yml

```diff
@@ -88,10 +89,12 @@ jobs:
           echo "Using runner: $RUNNER_NAME"
           echo "RUNNER_NAME=$RUNNER_NAME" >> $GITHUB_ENV
           echo "runnerName=$RUNNER_NAME" >> $GITHUB_OUTPUT
-
+  
+  # TODO: We use the same name for the workflow step and the step inside the workflow step which creates confusion
```

Comment:
Here, we have:
- one job
- one step
- an id for a step

They are all for the same thing.
Should we add suffixes to all of them, to make it clearer?
```yml
pick-environment-job
pick-environment-step
pick-environment-step-id
```
?

Reply:
This was refactored. Need to be re-reviewed 😅